proto/serve.proto

```diff
@@ -0,0 +1,224 @@
+    key: "default";
+    value: {
+      schema: {
+        json_schema: {
+          ref: ".api.Status";
+        }
+      }
+    }
+  }
+};
+
+service RayServeService {
+  // Create a new ray serve.
+  rpc CreateRayService(CreateRayServiceRequest) returns (RayService) {
+    option (google.api.http) = {
+      post: "/apis/v1alpha2/namespaces/{namespace}/services"
+      body: "service"
+    };
+  }
+  // Find a specific ray serve by name and namespace.
+  rpc GetRayService(GetRayServiceRequest) returns (RayService) {
+    option (google.api.http) = {
+      get: "/apis/v1alpha2/namespaces/{namespace}/services/{name}"
+    };
+  }
+  // Finds all ray services in a given namespace. Supports pagination, and sorting on certain fields.
+  rpc ListRayServices(ListRayServicesRequest) returns (ListRayServicesResponse) {
+    option (google.api.http) = {
+      get: "/apis/v1alpha2/namespaces/{namespace}/services"
+    };
+  }
+
+  // Finds all ray services in a given namespace. Supports pagination, and sorting on certain fields.
+  rpc ListAllRayServices(ListAllRayServicesRequest) returns (ListAllRayServicesResponse) {
+    option (google.api.http) = {
+      get: "/apis/v1alpha2/services"
+    };
+  }
+
+  // Deletes a ray service by its name and namespace 
+  rpc DeleteRayService(DeleteRayServiceRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/apis/v1alpha2/namespaces/{namespace}/services/{name}"
+    };
+  }
+}
+
+message CreateRayServiceRequest {
+  // The ray service to be created.
+  RayService service = 1;
+  // The namespace of the ray service to be created.
+  string namespace = 2;
+}
+
+message GetRayServiceRequest {
+  // The name used for retrieving the ray service.
+  string name = 1;
+  // The namespace used for retrieving the ray service.
+  string namespace = 2;
+}
+
+message ListRayServicesRequest {
+  // The namespace of the ray services to be retrieved.
+  string namespace = 1;
+  // A page token to request the next page of results. The token is acquried
+  // from the nextPageToken field of the response from the previous
+  // ListRayServices call or can be omitted when fetching the first page.
+  string page_token = 2;
+  // The number of RayServices to be listed per page. If there are more
+  // RayServices than this number, the response message will contain a
+  // nextPageToken field you can use to fetch the next page.
+  int32 page_size = 3;
+}
+
+message ListRayServicesResponse {
+  repeated RayService services = 1; 
+  // The total number of RayServices for the given query.
+  int32 total_size = 2;
+  // The token to list the next page of RayServices.
+  string next_page_token = 3;
+}
+
+message ListAllRayServicesRequest {
+  // A page token to request the next page of results. The token is acquried
+  // from the nextPageToken field of the response from the previous
+  // ListRayServices call or can be omitted when fetching the first page.
+  string page_token = 2;
```

Comment:
should it start with 1?

Reply:
> @scarlet25151
> 
> 1. Could you give a comparison between RayServe yaml and HTTP POST payload and that's easier to compare.
yes, I will add it to the top of the conversation.
> 2. Let's also make sure docs are updated.
this part is not updated yet, I will update in the third commit.
> 3. I do see few TODO items there, are they blockers for RayServe endpoint? If so, let's add the status summary in the PR description
here is one things I would like to discuss, should we use the raycluster's ingress or rayservice ingress here? https://github.com/ray-project/kuberay/blob/fdee883c3de7fa95fca6659bd200b98f2cc82b02/ray-operator/controllers/ray/rayservice_controller.go#L601-L603

Reply:
done.